tests/queries/0_stateless/01016_simhash_minhash.python

```diff
@@ -0,0 +1,394 @@
+#!/usr/bin/env python3                                                                                                                                                                                              
+import os
+import socket
+import sys
+from scipy import stats
+import pandas as pd
```

Comment:
More importantly, what I don't understand is why the expected results on PPC are different than on x86. Perhaps I missed something but was the whole point of the PR not to obtain CRC values that are equal across all platforms, given the same inputs?

If that is truly by design and cannot be changed, then there are better ways to test platform-specific behavior. Changing the SQL test into a Python test blows it up in a quite unaesthetic way. If you decide to go this route, then at least l. 28 - 183 should be factorized into a separate Python module, as they are copypasta from 02458_insert_select_progress_tcp and 02010_lc_native.python and at some point the dissemination of this boilerplate code needs to stop.

But there is no reason to mess with the native communication protocol at all for the purpose of testing CRCs. You could copy the original SQL test into two tests
- 01016_simhash_minhash.sql and
- 01016_simhash_minhash_ppc.sql

(and each SQL test will have a correspondingly named ".reference" file)

The first test is for all platforms except PPC, whereas the second test is just for PPC. Since they are copypasted, they will run the same queries (and ideally, they contain a comment at the top noting this fact and that both tests should be kept in-sync).

We can use the "Tags:" feature of SQL tests to restrict the platforms on which the 1st and 2nd tests will run. Search for this term in "tests/queries/0_stateless". Right now, tag "no-cpu-aarch64" is used in some places (to exclude ARM systems in CI). Tests in CI are executed by script "tests/clickhouse-test" and the code in there which evaluates the "no-" flag is
```py
elif tags:                                                        
│   for build_flag in args.build_flags:                           
│   │   if "no-" + build_flag in tags:                            
│   │   │   return FailureReason.BUILD                            
│   for tag in tags:                                              
│   │   tag = tag.replace("-", "_")                               
│   │   if tag.startswith("use_") and tag not in args.build_flags:
│   │   │   return FailureReason.BUILD                            
```

There are no PPC machines in ClickHouse's CI landscape (*). Basically, we want to
- exclude the 1st test from running on PPC (which is due to (*) an academic concern but nice to have) and
- exclude the 2nd test from running on x86 and ARM.

So the question is which tags must be supplied to both tests. For that, you could check clickhouse-test, specifically its method `collect_build_flags() and what is returned by ClickHouse's system table on PPC. I would like to help with that but I don't have easy access to a PPC machine.

Reply:
> More importantly, what I don't understand is why the expected results on PPC are different than on x86. Perhaps I missed something but was the whole point of the PR not to obtain CRC values that are equal across all platforms, given the same inputs?

Power's CRC32 vpmsum uses the  CRC table same as Intel and also uses the same polynomial. And the CRC table is the software fallback version and if the hardware has the crc32c, use the crc32-vpmsum for crc32 generation. Due to the difference in hardware, we see the different hash values.

Below is the transcript in telegram:
Meena Renganathan, [Dec 9, 2022 at 10:41:07 AM]:
It is CRC32C function for Power. The CRC table follows the same approach as Intel and also uses the same polynomial

Alexey Milovidov, [Dec 9, 2022 at 10:46:38 AM]:
Then why the results are different?

Meena Renganathan, [Dec 9, 2022 at 11:27:30 AM]:
The above table is the software fallback version and if the hardware has the crc32c, use the crc32-vpmsum for crc32 generation. May be due to the difference in hardware, we see the different hash values

Alexey Milovidov, [Dec 9, 2022 at 12:39:53 PM]:
Ok. ngram difference is an approximate function, let's consider the results on different platforms ok so far, but we can change it later.

Reply:
> We can use the "Tags:" feature of SQL tests to restrict the platforms on which the 1st and 2nd tests will run. Search for this term in "tests/queries/0_stateless". Right now, tag "no-cpu-aarch64" is used in some places (to exclude ARM systems in CI). Tests in CI are executed by script "tests/clickhouse-test" and the code in there which evaluates the "no-" flag is

Will start working on the above test approach.